cmd/vicadmin/vicadm.go

```diff
@@ -315,33 +333,36 @@ func findDatastoreLogs(c *session.Session) (map[string]entryReader, error) {
 	readers := map[string]entryReader{}
 	ctx := context.Background()
 
-	paths, err := listVMPaths(ctx, c)
+	logfiles, err := listVMPaths(ctx, c)
 	if err != nil {
 		detail := fmt.Sprintf("unable to perform datastore log collection due to failure looking up paths: %s", err)
 		log.Error(detail)
 		return nil, errors.New(detail)
 	}
 
-	for _, vmpath := range paths {
-		log.Debugf("Assembling datastore readers for %s", vmpath.String())
+	for _, logfile := range logfiles {
+		log.Debugf("Assembling datastore readers for %s", logfile.URL.String())
 		// obtain datastore object
-		ds, err := c.Finder.Datastore(ctx, vmpath.Host)
+		ds, err := c.Finder.Datastore(ctx, logfile.URL.Host)
 		if err != nil {
-			log.Errorf("Failed to acquire reference to datastore %s: %s", vmpath.Host, err)
+			log.Errorf("Failed to acquire reference to datastore %s: %s", logfile.URL.Host, err)
 			continue
 		}
 
 		// generate the full paths to collect
 		for _, file := range vmFiles {
 			// replace the VM token in file name with the VM name
-			processed := strings.Replace(file, string(vchconfig.VM), path.Base(vmpath.Path), -1)
-			rpath := fmt.Sprintf("%s/%s", vmpath.Path, processed)
-			readers[rpath] = datastoreReader{
+			vmHashName := strings.Split(logfile.VMName, "-")
+			processed := strings.Replace(file, string(vchconfig.VM), vmHashName[1], -1)
+			wpath := fmt.Sprintf("log write path: %s/%s", logfile.VMName, processed)
+			rpath := fmt.Sprintf("log read path: %s/%s", logfile.URL.Path, processed)
```

Comment:
Don't see how this can work with the "log read path: " prefix.  `datastoreReader.path` is used by datastoreReader.open in the GET request.  We can revisit the other name issues above, but this looks like it has to be fixed.